kotlinlib/src/mill/kotlinlib/KotlinModule.scala

```diff
@@ -0,0 +1,211 @@
+package mill
+package kotlinlib
+
+import mill.api.{PathRef, Result}
+import mill.define.{Command, Task}
+import mill.kotlinlib.worker.api.KotlinWorker
+import mill.scalalib.api.CompilationResult
+import mill.scalalib.{Dep, DepSyntax, JavaModule, Lib}
+import mill.{Agg, T}
+
+import java.io.File
+
+trait KotlinModule extends JavaModule with KotlinModulePlatform { outer =>
+
+  /**
+   * All individual source files fed into the compiler.
+   */
+  override def allSourceFiles = T {
+    Lib.findSourceFiles(allSources(), Seq("kt", "kts", "java")).map(PathRef(_))
+  }
+
+  /**
+   * All individual Java source files fed into the compiler.
+   * Subset of [[allSourceFiles]].
+   */
+  def allJavaSourceFiles = T {
+    allSourceFiles().filter(_.path.ext.toLowerCase() == "java")
+  }
+
+  /**
+   * All individual Kotlin source files fed into the compiler.
+   * Subset of [[allSourceFiles]].
+   */
+  def allKotlinSourceFiles = T {
+    allSourceFiles().filter(path => Seq("kt", "kts").exists(path.path.ext.toLowerCase() == _))
+  }
+
+  /**
+   * The Kotlin version to be used (for API and Language level settings).
+   */
+  def kotlinVersion: T[String]
+
+  /**
+   * The dependencies of this module.
+   * Defaults to add the kotlin-stdlib dependency matching the [[kotlinVersion]].
+   */
+  override def ivyDeps: T[Agg[Dep]] = T {
+    super.ivyDeps() ++ Agg(
+      ivy"org.jetbrains.kotlin:kotlin-stdlib:${kotlinVersion()}"
+    )
+  }

```

Comment:
Let's move this to `mandatoryIvyDeps`? That's where we put most things that are included by `trait`s and not intended to be removed by users. That would also let us remove the `super.ivyDeps()` from the examples, and bring them closer to the Java/Scala examples